internal/compiler-bridge/src/main/scala/xsbt/STJ.scala

```diff
@@ -0,0 +1,79 @@
+package xsbt
+import java.io.File
+import java.nio.file.Paths
+import java.util.zip.ZipFile
+
+class STJ(outputDirs: Iterable[File]) {
+  type JaredClass = String
+  type RelClass = String
+
+  def init(jar: File, cls: RelClass): JaredClass = {
```

Comment:
Could you add Scaladoc here, and perhaps consider renaming this method?
I normally use "init" to mean initialize STJ class, but it doesn't look like you're doing that.
Maybe a better name would be "jaredClassString"?

Reply:
Yes, that name is unfortunate. It was the first method I created when I started working on it and it kinda stayed like that for the whole time.
I think ideally for sake of readability I could have a value class `JaredClass` wrapping the string with method that extract stuff and companion object with factories. Though this is actually mostly kept as File I suspect there would be plenty of conversions. I will try to experiment with that and see if I can improve readability with that.

Reply:
As for scaladoc I will add it in places you suggested. But aside from that is there a general rule to put it e.g. on every public method I added or something like that, or just for less obvious code?

Reply:
I think it would be good to generally document public methods for Zinc project since many people are involved.